packages/kbn-search-errors/src/es_error.tsx

```diff
@@ -9,22 +9,32 @@
 import React from 'react';
 import { EuiButton, EuiCodeBlock, EuiSpacer } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-import { ApplicationStart } from '@kbn/core/public';
-import { KbnError } from '@kbn/kibana-utils-plugin/common';
+import type { ApplicationStart } from '@kbn/core/public';
 import { IEsError } from './types';
 import { getRootCause } from './utils';
 
-export class EsError extends KbnError {
+/**
+ * Checks if a given errors originated from Elasticsearch.
+ * Those params are assigned to the attributes property of an error.
+ *
+ * @param e
+ */
+export function isEsError(e: any): e is IEsError {
+  return !!e.attributes;
+}
+
+export class EsError extends Error {
   readonly attributes: IEsError['attributes'];
 
   constructor(protected readonly err: IEsError, private readonly openInInspector: () => void) {
     super(
       `EsError: ${
         getRootCause(err?.attributes?.error)?.reason ||
-        i18n.translate('data.esError.unknownRootCause', { defaultMessage: 'unknown' })
+        i18n.translate('searchErrors.esError.unknownRootCause', { defaultMessage: 'unknown' })
       }`
     );
     this.attributes = err.attributes;
+    Object.setPrototypeOf(this, new.target.prototype);
```

Comment:
What does this do?

Reply:
Is this equivalent to `Object.setPrototypeOf(this, EsError.prototype);`? If so, I think it would read better if we changed it to that.